lib/cfn-model/psych/handlers/line_number_handler.rb

```diff
@@ -0,0 +1,18 @@
+# We need to provide a handler that will add the line to the node
+# as it is parsed. TreeBuilder is the "usual" handler, that
+# creates the AST.
+class LineNumberHandler < Psych::TreeBuilder
+
+  # The handler needs access to the parser in order to call mark
+  attr_accessor :parser
+
+  # We are only interested in scalars, so here we override
+  # the method so that it calls mark and adds the line info
+  # to the node.
+  def scalar value, anchor, tag, plain, quoted, style
+    mark = parser.mark
+    s = super
```

Comment:
is Node somewhere up in the inheritance hierarchy (and that's what 's' is?)  the name 's' is a little confusing... i guess short for 'super' but what is it really?  i *think* i get it, but any elaboration here on the order of execution during parsing could be helpful in understanding

Reply:
Its actually scalar. I'll change it to `scalar_value`